spyder/workers/updates.py

```diff
@@ -238,13 +236,39 @@ def _download_installer(self):
 
         installer_path = osp.join(installer_dir_path, name)
         self.installer_path = installer_path
-        if not osp.isfile(installer_path):
-            logger.debug(
-                f"Downloading installer from {url} to {installer_path}")
-            urlretrieve(
-                url, installer_path, reporthook=self._progress_reporter)
-        else:
-            self._progress_reporter(1, 1)
+
+        if osp.isfile(installer_path):
+            # Installer already downloaded
+            logger.info(f"{installer_path} already downloaded")
+            self._progress_reporter(1, 1, 1)
+            return
+
+        logger.debug(f"Downloading installer from {url} to {installer_path}")
+        with (
+            requests.get(url, stream=True) as r,
```

Comment:
@ccordoba12 @CAM-Gerlach , this line is causing a syntax error for only Linux and Python 3.8 CI tests. Win and macOS seem to be okay. Any thoughts?